### PR TITLE
Add robotiq f2 140 support

### PR DESCRIPTION
**Pull request by JafarAbdi**
Copied from git@github.com:ros-industrial/robotiq.git with small modifications to have the same structure as the codebase here

TODOs:

- [x] Update ros2_control file to support f2_140
- [x] Update launch files to supoort f2_140

### CODE REVIEW

**Review by MarqRazz — DISMISSED** [review]
I don't have my computer to test but the changes look good. Thanks for adding the 2f_140!

File: robotiq_description/launch/view_gripper.launch.py
Comment (MarqRazz) [inline robotiq_description/launch/view_gripper.launch.py:44]: This should be `robotiq_2f_85.urdf.xacro`

File: robotiq_description/urdf/robotiq_2f_85_macro.urdf.xacro
Comment (MarqRazz) [inline robotiq_description/urdf/robotiq_2f_85_macro.urdf.xacro:36]: You missed adding `2f_85` to all of the visual tags.

File: robotiq_description/urdf/robotiq_arg2f_140.xacro
Comment (MarqRazz) [inline robotiq_description/urdf/robotiq_arg2f_140.xacro:1]: What is this file used for?

**Review by MarqRazz — COMMENTED** [review]
Sorry I forgot about this PR! I don't have a way to test with hardware but the `view_gripper` launch files work.

image

Can you change the 2f_140 xacro's to mach how the 2f85 was done? 
That is `robotiq_2f_85.urdf.xacro` calls the macro and attached to a world link. The `macro.urdf.xacro` includes ros2_control and builds the gripper URDF (feel free to keep that xacro only file).

**Comment by develiberta** [comment]
I'm just waiting for this request to be merged.

**Comment by JafarAbdi** [comment]
> I'm just waiting for this request to be merged.

I'll fix the conflicts and try to get this working this week

**Comment by skpawar1305** [comment]
thank you
gripper works with gen3

by adding 
```
<?xml version="1.0"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro" name="robotiq_gripper">
    <xacro:macro name="robotiq_gripper" params="
        name
        prefix
        parent
        *origin
        sim_ignition:=false
        sim_isaac:=false
        use_fake_hardware:=false
        fake_sensor_commands:=false
        include_ros2_control:=true
        com_port:=/dev/ttyUSB0">

        <!-- ros2 control include -->
        <xacro:include filename="$(find robotiq_description)/urdf/2f_140.ros2_control.xacro" />
        <!-- if we are simulating or directly communicating with the gripper we need a ros2 control instance -->
        <xacro:if value="${include_ros2_control}">
            <xacro:robotiq_gripper_ros2_control
                name="${name}" prefix="${prefix}"
                sim_ignition="${sim_ignition}"
                sim_isaac="${sim_isaac}"
                use_fake_hardware="${use_fake_hardware}"
                fake_sensor_commands="${fake_sensor_commands}"
                com_port="${com_port}"/>
        </xacro:if>

        <xacro:include filename="$(find kortex_description)/grippers/robotiq_2f_140/urdf/robotiq_arg2f_140_model_macro.xacro" />
        <xacro:robotiq_arg2f_140 parent="${parent}" prefix="${prefix}"/>
    </xacro:macro>
</robot>
```
to a new file robotiq_2f_140_macro.urdf.xacro
and renaming gripper and joint in gen3's launch file

**Review by MarqRazz — COMMENTED** [review]
I have verified this works in Gazebo and with the Kinova URDF (I will open a PR after we get this merged).
image

One issue that I have is that the 2f-85 in this repo has the fingers moving along the X-axis of the gripper base...
image

and the 2f-140 has the fingers moving along they Y-axis so it requires the user add an additional rotation when switching between the two.
image

To properly fix this/make them match we would need to rotate some of the meshes and then fix all the joints. Should we block the PR until I have time to fix the rotation?

**Review by MarqRazz — APPROVED** [review]
Okay so I added a dummy link to the base of the 2f-140 gripper to match the 2f-85. I will open up an issue to document the extra link and if someone has time they can rotate the meshes and delete it so they match.

**Review by moriarty — APPROVED** [review]

**Pull request merged**
